pkgs/by-name/de/deadlock-modmanager/package.nix

```
@@ -0,0 +1,160 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  cargo-tauri,
+  nodejs,
+  pnpm_9,
+  pkg-config,
+  wrapGAppsHook3,
+  desktop-file-utils,
+  webkitgtk_4_1,
+  cairo,
+  gdk-pixbuf,
+  glib,
+  glib-networking,
+  gtk3,
+  libsoup_3,
+  pango,
+  openssl,
+  bzip2,
+  gst_all_1,
+  makeDesktopItem,
+  fontconfig,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "deadlock-modmanager";
+  version = "0.9.2";
+
+  src = fetchFromGitHub {
+    owner = "Stormix";
+    repo = "deadlock-modmanager";
+    tag = "v${version}";
+    hash = "sha256-meCwpdnRYwNiOB8QzdhDC24CtRiMClimCDlv0EvHHgI=";
+  };
+
+  cargoRoot = "apps/desktop";
+
+  cargoHash = "sha256-cLdsR9ZCkUKZQDAxsvZpau7c9LxeLt4kUI21ObiY7dA=";
+
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
+    cargo-tauri
+    nodejs
+    pnpm_9.configHook
+    pkg-config
+    wrapGAppsHook3
+  ];
+
+  buildInputs = [
+    webkitgtk_4_1
+    cairo
+    gdk-pixbuf
+    glib
+    glib-networking
+    gtk3
+    libsoup_3
+    pango
+    openssl
+    bzip2
+    desktop-file-utils
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-bad
+  ];
+
+  pnpmRoot = ".";
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit
+      pname
+      version
+      src
+      ;
+    fetcherVersion = 2;
+    sourceRoot = "source";
+    hash = "sha256-ZnVUQjPXphwk7J5jd04WQdUYQF0t5veqRVr/m4guuEU=";
+  };
+
+  VITE_API_URL = "https://api.deadlockmods.app";
+
+  buildPhase = ''
+    runHook preBuild
+
+    cd apps/desktop
+    export CARGO_TARGET_DIR=target
+
+    # Then build the Tauri app
+    cargo tauri build --no-bundle -- --frozen
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # Debug: show what's in the current directory
+    echo "Contents of apps/desktop:"
+    ls -la
+    echo "Contents of dist (if exists):"
+    ls -la dist/ || echo "No dist directory"
+
+    # Install the binary (we're still in apps/desktop from buildPhase)
+    install -Dm755 src-tauri/target/release/deadlock-mod-manager $out/bin/${pname}
+
+    # Install icons
+    install -Dm644 src-tauri/icons/32x32.png $out/share/icons/hicolor/32x32/apps/${pname}.png
+    install -Dm644 src-tauri/icons/128x128.png $out/share/icons/hicolor/128x128/apps/${pname}.png
+    install -Dm644 src-tauri/icons/128x128@2x.png $out/share/icons/hicolor/256x256/apps/${pname}.png
+
+    # Install web assets
+    if [ -d "dist" ]; then
+      mkdir -p $out/share/${pname}
+      cp -r dist $out/share/${pname}/
+      echo "Installed dist directory"
+    else
+      echo "Warning: dist directory not found"
+    fi
```

```suggestion
    mkdir -p $out/share/deadlock-modmanager
    cp -r dist $out/share/deadlock-modmanager/
```
we most likely want to fail here if it does not exist